rpcserver.go

```diff
@@ -2278,6 +2312,68 @@ func (r *rpcServer) ListPeers(ctx context.Context,
 	return resp, nil
 }
 
+// GetFeatures returns the features that a peer supports.
+func (r *rpcServer) GetFeatures(ctx context.Context,
+	in *lnrpc.GetFeaturesRequest) (*lnrpc.GetFeaturesResponse, error) {
+
+	if len(in.PubKey) == 0 {
+		return nil, fmt.Errorf("pubkey must be provided")
+	}
+
+	// If the query is for our node, get our set of enabled features.
+	if bytes.Equal(in.PubKey, r.selfNode[:]) {
```

Comment:
we could also consider adding these features to `getinfo` so what we don't have to query with our own pubkey